src/app/utils/Encoder.ts

```diff
@@ -1,58 +1,59 @@
 import { JSONValue, JSONValueWithoutArray, Utils } from 'src/app/utils/utils';
-import { assert } from 'src/app/utils/assert';
 
-export abstract class Encoder<T> {
-    public abstract encode(move: T): JSONValue;
+export abstract class AbstractEncoder<T> {
 
-    public abstract decode(encodedMove: JSONValue): T;
+    public abstract encodeValue(move: T): JSONValue;
+
+    public abstract decodeValue(encodedMove: JSONValue): T;
 }
 
 // Used internally. If T = [A, B, C], then
 // EncoderArray<T> = [Encoder<A>, Encoder<B>, Encoder<C>]
-type EncoderArray<T> = { [P in keyof T]: Encoder<T[P]> };
+type EncoderArray<T> = { [P in keyof T]: AbstractEncoder<T[P]> };
+
+export abstract class Encoder<T> extends AbstractEncoder<T> {
```

Comment:
L'idée de MoveEncoder c'est que c'était un Encoder limité pour encoder les moves, on n'en a plus besoin car on stocke plus les mouvements en array, donc peut-être simpement juste le jarter ? Et garder l'ancienne class Encoder. Y'a la distinction JSONValueWithoutArray/JSONValue à faire disparaître des encodeurs du coup, le reste est adaptable facilement à partir de là

Reply:
AbstractEncoder is dead !